client/src/search/Typeahead/Typeahead.js

```diff
@@ -0,0 +1,215 @@
+
+  render() {
+    const {
+      placeholder,
+      minLength,
+      filter_content,
+      pagination_size,
+      search_configs,
+      onNewQuery,
+    } = this.props;
+
+    const { search_text, can_show_menu } = this.state;
+
+    const debounceOnNewQuery = _.debounce((query) => {
+      onNewQuery();
+      log_standard_event({
+        SUBAPP: window.location.hash.replace("#", ""),
+        MISC1: `TYPEAHEAD_SEARCH_QUERY`,
+        MISC2: `query: ${query}, search_configs: ${_.map(
+          search_configs,
+          "config_name"
+        )}`,
+      });
+    }, 500);
+
+    const update_search_text = (event) => {
+      const text = _.trimStart(event.target.value); //prevent empty searching that will show all results
+      debounceOnNewQuery(text);
+      this.setState({ search_text: text });
+    };
+
+    const config_groups = _.map(search_configs, (search_config, ix) => ({
+      group_header: search_config.header_function(),
+      group_filter: search_config.filter,
+    }));
+
+    // Options includes placeholders for pagination items, because the number of results passed to renderMenu
+    // (ie. that get through matches_query) needs to actually match the number of lis ultimately rendered, can't
+    // just insert the pagination items when renderMenu is called
+    const all_options = [
+      ..._.flatMap(search_configs, (search_config, ix) =>
+        _.map(search_config.get_data(), (data) => ({
+          data,
+          name: search_config.name_function(data),
+          menu_content: (search) =>
+            _.isFunction(search_config.menu_content_function) ? (
+              search_config.menu_content_function(data, search)
+            ) : (
+              <InfoBaseHighlighter
+                search={search}
+                content={search_config.name_function(data)}
+              />
+            ),
+          config_group_index: ix,
+        }))
+      ),
+    ];
+
+    const matches_query = (option) => {
+      const query = this.state.search_text;
+      const group_filter =
+        config_groups[option.config_group_index].group_filter;
+      const query_matches = group_filter(query, option.data);
+
+      return query_matches;
+    };
+
+    const queried_results = _.filter(
+      all_options,
+      (res) => matches_query(res) && !_.isUndefined(res.config_group_index)
+    );
+
+    const menu_props = {
+      queried_results,
+      pagination_size,
+      search_text,
+      config_groups,
+      on_select_item: this.on_select_item,
+      hide_menu: this.hide_menu,
+    };
+
+    return (
+      <div className="rbt" style={{ position: "relative" }}>
+        <div className="search-bar">
+          <div className="search-icon-container">
+            <span aria-hidden="true">
+              <img
+                src={`${get_static_url("svg/search.svg")}`}
+                style={{ width: "30px", height: "30px" }}
+              />
+            </span>
+          </div>
+          <input
```

Comment:
Can you add `aria-label` to this input?🙂 Just describing that this is basically for searching